### PR TITLE
Merges 2 tool bar add buttons that use dropdowns

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
this addresses #8361.  We had essentially duplicated components so this removes the 1 that was being used inside some of the Inventory tree in favor of the 1 inside of the components tree.  No functionality is impacted.
##### ISSUE TYPE

-enhancement

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
Screen Shot 2020-10-21 at 3 13 40 PM

### CODE REVIEW

**Comment by unlikelyzero** [comment]
@AlexSCorey please give us some time to update our tests to match this new layout

**Review by jakemcdermott — CHANGES_REQUESTED** [review]
I like this PR. Combining these components into one is a good direction to go in. :+1: 

We probably don't need a configuration object or a generator function for the add button. After looking at this w/ @AlexSCorey the plan is to update the component such that it just accepts a list of DropDownItems that the caller can define explicitly (much like the standard PF DropDown component, which this wraps).

example here: https://github.com/jakemcdermott/awx/commit/9e4fb3c18617a276c4a792cca2ee10a93ff0227e

File: awx/ui_next/package.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
   "scripts": {
     "start": "PORT=3001 HTTPS=true DANGEROUSLY_DISABLE_HOST_CHECK=true react-scripts start",
     "build": "react-scripts build",
-    "test": "TZ='UTC' react-scripts test --coverage --watchAll=false",
+    "test": "TZ='UTC' react-scripts test --watchAll=false",
```
Comment (jakemcdermott) [inline awx/ui_next/package.json]:
```suggestion
    "test": "TZ='UTC' react-scripts test --coverage --watchAll=false",
```

File: awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx [hunk 2]
```diff
@@ -135,24 +134,35 @@ function TemplateList({ i18n }) {
...
-
-  if (canAddWFJT) {
-    addButtonOptions.push({
-      label: i18n._(t`Workflow Template`),
-      url: `/templates/workflow_job_template/add/`,
-    });
-  }
 
+  const addTempate = toTitleCase(i18n._(t`Add Job Template`));
+  const addWFTemplate = toTitleCase(i18n._(t`Add Job Template`));
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx]: Screenshot from 2020-10-23 09-48-44
```suggestion
  const addWFTemplate = toTitleCase(i18n._(t`Add Workflow Template`));
```

File: awx/ui_next/src/components/AddDropDownButton/AddDropDownButton.test.jsx [hunk 3]
```diff
@@ -23,7 +21,7 @@ describe('<AddDropDownButton />', () => {
     );
     wrapper.find('button').simulate('click');
     expect(wrapper.find('Dropdown').prop('isOpen')).toEqual(true);
-    expect(wrapper.find('Link')).toHaveLength(dropdownItems.length);
+    expect(wrapper.find('DropdownItem')).toHaveLength(dropdownItems.length);
   });
```
Comment (jakemcdermott) [inline awx/ui_next/src/components/AddDropDownButton/AddDropDownButton.test.jsx:26]: @AlexSCorey 

Let's add a test to check the kebab-related behavior.

**Comment by jakemcdermott** [comment]
@AlexSCorey This is looking great. Just a few more things to do and this will be ready for QE:

- Misc. small fixups (I left them as edit suggestions in the review comments)
- Need a test case for kebab-related code paths in the component. This should help w/ test coverage considerably.
- Small merge conflict w/ latest changes to `TemplateList` on devel

Once these things are addressed I am :100: :+1: :ship: on this PR. Nice work!

File: awx/ui_next/package.json (hunk 1 above)
Comment (AlexSCorey) [reply awx/ui_next/package.json]: whoops, meant to remove that.

**Comment by unlikelyzero** [comment]
@tiagodread will be starting on tests for this tomorrow

**Review by jakemcdermott — APPROVED** [review]

**Comment by tiagodread** [comment]
Hey @AlexSCorey I found some places where we are using button inside kebab options:

**Job Templates:**

- Access tab -> advanced search -> kebab -> Add

image

- Access tab -> Add -> Users or team -> advanced search -> kebab -> empty (same happen in WFJT Node) 

image

**Credentials:**

- Access tab -> advanced search -> kebab -> Add

image

**Projects:**

- Access tab -> advanced search -> kebab -> Add

image
- Notifications -> advanced search -> kebab -> empty

image

**Inventories**

- Access tab -> advanced search -> kebab -> Add

image

- Groups tab -> advanced search -> kebab -> delete

image

**Hosts**

- Groups tabs -> advanced search -> kebab -> Add

image

**Organizations**

- Access tab -> advanced search -> kebab -> Add

image

**Teams**

- Access tab -> advanced search -> kebab -> Add

image

- Roles tab -> advanced search -> kebab -> Add

image

**Instance groups**

- Instance tab -> advanced search -> kebab -> Associate
image

**Comment by AlexSCorey** [comment]
> Hey @AlexSCorey I found some places where we are using button inside kebab options:

I think all of these issues should be documented and tracked in https://github.com/ansible/awx/issues/8237.

**Comment by tiagodread** [comment]
Moved above items to 8237.

@AlexSCorey I added a test case where the user can add only a job template (a normal user with project admin) on devel the dropdown button shows only one option "Add job template" (correct) on this branch I can see both options JT and WFJT we have 2 failing ui tests.

**Review by tiagodread — APPROVED** [review]
👍🏽

**Pull request merged**